File: solutions/lua/resistor-color-trio/1/resistor-color-trio.lua
return {
  decode = function(c1, c2, c3)
  colors={'black','brown','red','orange',
        'yellow','green','blue','violet',
        'grey','white'}
    for i=1,10,1
    do
      if colors[i]==c1
      then 
        value = 10 * (i-1)
        break
      end
    end  
    for i=1,10,1
    do
      if colors[i]==c2
      then 
        value = value + (i-1)
        break
      end
    end
    for i=1,10,1
    do
      if colors[i]==c3
      then 
        pow = i-1 
        break
      end
    end
    for i = 1,pow,1
    do
      value=value * 10
    end  

    if value % 1000 == 0 
    then 
      unit='kiloohms'
      value=value/1000
    else  
      unit='ohms'
    end  
    return value, unit
  end
}

File: solutions/julia/resistor-color-trio/1/resistor-color-trio.jl
function label(colors)
    cv=Dict("black"=> "0","brown"=> "1","red"=> "2",
    "orange"=> "3","yellow"=> "4","green"=> "5",
    "blue"=> "6","violet"=> "7","grey"=> "8","white"=> "9")
    r=""
    for i in 1:length(colors)-1
        r=r * cv[colors[i]]
    end
    r=r*repeat("0",parse(Int32,cv[colors[end]]))*" ohms"
    ra=findfirst("000 ohms",r)
    if !isnothing(ra)
       r = r[1:first(ra)-1]*" kiloohms"
    end
    return r
end
